rush-plugins/rush-azure-storage-build-cache-plugin/src/RushAzureInteractiveAuthPlugin.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { IRushPlugin, RushSession, RushConfiguration, ILogger } from '@rushstack/rush-sdk';
+import type { AzureEnvironmentNames } from './AzureStorageBuildCacheProvider';
+
+const PLUGIN_NAME: 'AzureInteractiveAuthPlugin' = 'AzureInteractiveAuthPlugin';
+
+/**
+ * @public
+ */
+export interface IAzureInteractiveAuthOptions {
+  /**
+   * The name of the the Azure storage account to use for build cache.
+   */
+  readonly storageAccountName: string;
+
+  /**
+   * The name of the container in the Azure storage account to use for build cache.
+   */
+  readonly storageContainerName: string;
+
+  /**
+   * The Azure environment the storage account exists in. Defaults to AzureCloud.
+   */
+  readonly azureEnvironment?: AzureEnvironmentNames;
+
+  /**
+   * The set of Rush global commands before which credentials should be updated.
+   */
+  readonly globalCommands?: string[];
+
+  /**
+   * The set of Rush phased commands before which credentials should be updated.
+   */
+  readonly phasedCommands?: string[];
+}
+
+/**
+ * @public
+ */
+export default class RushAzureInteractieAuthPlugin implements IRushPlugin {
+  private readonly _options: IAzureInteractiveAuthOptions;
+
+  public readonly pluginName: 'AzureInteractiveAuthPlugin' = PLUGIN_NAME;
+
+  public constructor(options: IAzureInteractiveAuthOptions) {
+    this._options = options;
+  }
+
+  public apply(rushSession: RushSession, rushConfig: RushConfiguration): void {
+    const options: IAzureInteractiveAuthOptions = this._options;
+
+    const { globalCommands, phasedCommands } = options;
+
+    const { hooks } = rushSession;
+
+    const handler: () => Promise<void> = async () => {
+      const { AzureStorageBuildCacheProvider } = await import('./AzureStorageBuildCacheProvider');
+
+      const logger: ILogger = rushSession.getLogger(PLUGIN_NAME);
+      logger.terminal.writeLine(`Refreshing credentials`);
+
+      await new AzureStorageBuildCacheProvider({
+        storageAccountName: options.storageAccountName,
+        storageContainerName: options.storageContainerName,
+        azureEnvironment: options.azureEnvironment,
+        isCacheWriteAllowed: true
+      }).updateCachedCredentialInteractiveAsync(logger.terminal);
```

Comment:
This doesn't check if the credentials are already present and non-expired. This should probably only happen if the credentials actually need to be updated.

Reply:
Added a configurable minimum expiration time. If not specified will always renew.